### PR TITLE
Fix logical issues in Kubernetes Deployment Provider

### PR DESCRIPTION
**Pull request by imesh**
This pull request fixes following issues:
- Add namespace to DeploymentProvider interface and its implementations
- Add exception mappers to the application
- Uplift fabric8 kubernetes api version to 2.2.205

### CODE REVIEW

File: components/org.wso2.carbon.multitenancy.deployment.service/src/test/java/org/wso2/carbon/multitenancy/deployment/service/tests/MockDeploymentProvider.java [hunk 1]
```diff
@@ -29,39 +29,54 @@
...
         }
 
-        if (deploymentMap.get(deployment.getId()) != null) {
-            deploymentMap.remove(deployment.getId());
+        Map<String, Deployment> deploymentsMap = namespaceDeploymentsMap.get(namespace);
+        if (deploymentsMap == null) {
+            throw new RuntimeException("Deployment [" + deployment + "] not found in namespace " + namespace);
+        }
+
+        if (deploymentsMap.get(deployment.getId()) != null) {
```
Comment (jsdjayanga) [inline components/org.wso2.carbon.multitenancy.deployment.service/src/test/java/org/wso2/carbon/multitenancy/deployment/service/tests/MockDeploymentProvider.java]: why do we need to check this? I believe we can directly remove

File: components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java [hunk 2]
```diff
@@ -96,11 +106,18 @@ public KubernetesDeploymentProvider() {
...
      * @return Array of deployments
      */
     @Override
-    public List<Deployment> listDeployments() {
-        return kubernetesClient.extensions().deployments().list().getItems().stream()
+    public List<Deployment> listDeployments(String namespace) {
+        DeploymentList list = kubernetesClient.extensions().deployments().inNamespace(namespace)
+                .withLabel(WSO2_DEPLOYMENT_LABEL).list();
+        if (list == null || list.getItems() == null || list.getItems().isEmpty()) {
+            return new ArrayList<>();
```
Comment (jsdjayanga) [inline components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java]: Does the caller of this method expects to add new elements to the returned List. If not we can use Collections.emptyList();

File: components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java [hunk 3]
```diff
@@ -144,61 +164,149 @@ public Deployment getDeployment(String id) throws DeploymentNotFoundException {
...
+                            for (HTTPIngressPath path : rule.getHttp().getPaths()) {
+                                if (path.getBackend() != null) {
+                                    moveStringValToIntVal(path.getBackend().getServicePort());
+                                }
+                            }
+                        }
+                    }
+                }
+
+                logger.debug("Creating ingress: " + ingress.getMetadata().getName());
```
Comment (jsdjayanga) [inline components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java]: debug log check is needed as this is;
1. inside a loop 
2. A method call and string contact is happening all the time

File: components/org.wso2.carbon.multitenancy.deployment.service/src/test/java/org/wso2/carbon/multitenancy/deployment/service/tests/MockDeploymentProvider.java (hunk 1 above)
Comment (imesh) [inline components/org.wso2.carbon.multitenancy.deployment.service/src/test/java/org/wso2/carbon/multitenancy/deployment/service/tests/MockDeploymentProvider.java]: Thanks Jayanga! Will remove this!

File: components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java (hunk 2 above)
Comment (imesh) [inline components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java]: Thanks for pointing! Will update!

**Pull request merged**